substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -725,3 +725,126 @@ pub mod v5 {
 		}
 	}
 }
+
+/// This migration accumulates and initializes the [`TotalValueLocked`] for all pools.
+pub mod v6 {
+	use super::*;
+
+	/// [`VersionUncheckedMigrateV5ToV6`] wrapped in a
+	/// [`frame_support::migrations::VersionedMigration`], ensuring the migration is only
+	/// performed when on-chain version is 5.
+	#[cfg(feature = "experimental")]
+	pub type VersionCheckedMigrateV5ToV6<T> = frame_support::migrations::VersionedMigration<
+		5,
+		6,
+		VersionUncheckedMigrateV5ToV6<T>,
+		crate::pallet::Pallet<T>,
+		<T as frame_system::Config>::DbWeight,
+	>;
+
+	pub struct VersionUncheckedMigrateV5ToV6<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> VersionUncheckedMigrateV5ToV6<T> {
+		fn calculate_tvl_by_total_stake() -> BalanceOf<T> {
+			BondedPools::<T>::iter()
```

Comment:
fyi the typical attack on this is for an attacker to wait until this migration is _just about to happen_, then inflate this map. 

In Polkadot and Kusama this cannot happen, but you can still write it as a warning about the assumptions of this migration.

Reply:
I added a warning to the code. Thanks for pointing it out.